sdk/storage/storage-blob/src/RetryPolicyFactory.ts

```diff
@@ -5,6 +5,7 @@ import { RequestPolicy, RequestPolicyFactory, RequestPolicyOptions } from "@azur
 import { RetryPolicy, RetryPolicyType } from "./policies/RetryPolicy";
 
 export { RetryPolicyType } from "./policies/RetryPolicy";
+export { RetryPolicy };
```

Comment:
Why is this being exported? 
From what I can see, the `RetryPolicy` is used only in the `RetryPolicyFactory` which in turn is internal to the library and not exposed to the user

Reply:
The RetryPolicyFactory/RetryPolicy should be exported as it is useful when composing a custom pipeline.

Reply:
Renamed to `StorageBlobRetryPolicy` since storage queue exports its own retry policy as well.

Reply:
Rename is part of #5862